### PR TITLE
ci: filter for docs check to detect 3rd person("you, they")

### LINKED ISSUE #13907: Add a filter for docs check to detect 3rd person("you, they")
In the docs we try not to use "you"

We can simply check this in the CI, so add a check to not use 3rd person
_Originally posted by @travisn in https://github.com/rook/rook/pull/13821#discussion_r1518267910_

### PR DESCRIPTION
**Pull request by mmaoyu**
Check the docs to not use 3rd person under directory Documentation. 
List all added or changed files, and then use shell command 'grep' filter if the keywords present.

Resolves #13907

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/docs-keywords-filter.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        run: |
+
+          echo -e "Changed files: \n ${ALL_CHANGED_DOCS}"
+
+          # Get SHA
+          if [[ "${{ github.event_name }}" == "pull_request" ]];then
+             base_sha=${{ github.event.pull_request.base.sha }}
+             head_sha=${{ github.event.pull_request.head.sha }}
+          else
+            base_sha=${{ github.event.before }}
```
Comment (parth-gr) [inline .github/workflows/docs-keywords-filter.yml:56]: what does this do?

File: .github/workflows/docs-keywords-filter.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            # Delete diff summary by sed
+            # Filter added/modified lines , which with a '+' prefix
+            # Match keywords as words and ignore case
+            content_with_keyword="$(git diff ${base_sha} ${head_sha} ${file}|sed '0,/^@/d'|grep '^+'|grep -iw ${GREP_KEYWORD_DICS})"
+            if [[ "x${content_with_keyword}" != "x" ]] ; then
+              error_msg="${error_msg}${file}\n${content_with_keyword}\n\n"
+            fi
+          done
+
+          if [[ "x${error_msg}" != "x" ]]; then
```
Comment (parth-gr) [inline .github/workflows/docs-keywords-filter.yml:77]: can we move this to the above check itself

File: .github/workflows/docs-keywords-filter.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          files: |
+            Documentation/**.md
+            Documentation/**/**.md
+
+      - name: List all changed docs with keywords
+        if: steps.changed-docs.outputs.any_changed == 'true'
+        env:
+          ALL_CHANGED_DOCS: ${{ steps.changed-docs.outputs.all_changed_files }}
+          # Keywords regex for grep
+          GREP_KEYWORD_DICS: 'you\|they'
```
Comment (parth-gr) [inline .github/workflows/docs-keywords-filter.yml:44]: @travisn do you think of some other keywords

File: .github/workflows/docs-keywords-filter.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          error_msg=''
+          # Grep may return 1 when no added content
+          set +e
+
+          for file in ${ALL_CHANGED_DOCS}; do
+            # Get changed content by 'git diff'
+            # Delete diff summary by sed
+            # Filter added/modified lines , which with a '+' prefix
+            # Match keywords as words and ignore case
+            content_with_keyword="$(git diff ${base_sha} ${head_sha} ${file}|sed '0,/^@/d'|grep '^+'|grep -iw ${GREP_KEYWORD_DICS})"
```
Comment (parth-gr) [inline .github/workflows/docs-keywords-filter.yml:71]: can you add a second temporary commit with a document  change having `you` init, to see if this works properly

**Review by parth-gr — CHANGES_REQUESTED** [review]

File: .github/workflows/docs-keywords-filter.yml (hunk 1 above)
Comment (mmaoyu) [reply .github/workflows/docs-keywords-filter.yml:56]: This is for push event to get sha

File: .github/workflows/docs-keywords-filter.yml (hunk 4 above)
Comment (mmaoyu) [reply .github/workflows/docs-keywords-filter.yml]: I've added,and the actions result correct
[https://github.com/rook/rook/actions/runs/8325155167/job/22778182972](url)

File: .github/workflows/docs-keywords-filter.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
+name: docs-keywords-filter
```
Comment (subhamkrai) [inline .github/workflows/docs-keywords-filter.yml:1]: @mmaoyu Thanks for PR this looks good, one suggestion is could make this action as part of `doc-check.yml` CI.

IMO, this should be called as a part of `doc-check.yaml` Ci test?

**Review by travisn — COMMENTED** [review]
Honestly, I am not sure it is worth the effort to enforce this in the CI. There are currently over 500 instances of the word "you" in the Documentation folder. It will take a lot of effort to change all of those instances and rephrase it to something better. 

In the past year we have started reviewing the docs and rephrasing these usages of "you", however there is a lot of documentation still to review. We don't have a timeline to complete that review since other priorities commonly come up. Even when that review is completed, we may find some instances that we want to keep "you", instead of banning it completely from the docs. While I appreciate the effort to clean this up and automate the checking in the CI, I would suggest we close this. There are many other work items to consider higher priority, such as the issues found in the latest project boards. Thanks!

**Comment by parth-gr** [comment]
> Honestly, I am not sure it is worth the effort to enforce this in the CI. There are currently over 500 instances of the word "you" in the Documentation folder. It will take a lot of effort to change all of those instances and rephrase it to something better.
> 
> In the past year we have started reviewing the docs and rephrasing these usages of "you", however there is a lot of documentation still to review. We don't have a timeline to complete that review since other priorities commonly come up. Even when that review is completed, we may find some instances that we want to keep "you", instead of banning it completely from the docs. While I appreciate the effort to clean this up and automate the checking in the CI, I would suggest we close this. There are many other work items to consider higher priority, such as the issues found in the latest project boards. Thanks!

SO with this change we do not need to check the old instances, it is to not to add in future, 
If you see closely we have `git diff` added so which checks for the changes only done with the PR

**Comment by travisn** [comment]
> SO with this change we do not need to check the old instances, it is to not to add in future, If you see closely we have `git diff` added so which checks for the changes only done with the PR

I see, the PR is only checking for new instances. But even with that, I'm still not sure we can completely ban these words because it may depend on the context. The words may still be valid in some situations.

**Comment by Madhu-1** [comment]
> > SO with this change we do not need to check the old instances, it is to not to add in future, If you see closely we have `git diff` added so which checks for the changes only done with the PR
> 
> I see, the PR is only checking for new instances. But even with that, I'm still not sure we can completely ban these words because it may depend on the context. The words may still be valid in some situations.

i agree on this one, it will become very difficult to have this strict check for the documents.

**Comment by mmaoyu** [comment]
> > > SO with this change we do not need to check the old instances, it is to not to add in future, If you see closely we have `git diff` added so which checks for the changes only done with the PR
> > 
> > 
> > I see, the PR is only checking for new instances. But even with that, I'm still not sure we can completely ban these words because it may depend on the context. The words may still be valid in some situations.
> 
> i agree on this one, it will become very difficult to have this strict check for the documents.

I agree . Thanks to all the replies .
Since there is no need to do that, I will close this pr

**Comment by parth-gr** [comment]
> > > SO with this change we do not need to check the old instances, it is to not to add in future, If you see closely we have `git diff` added so which checks for the changes only done with the PR
> > 
> > 
> > I see, the PR is only checking for new instances. But even with that, I'm still not sure we can completely ban these words because it may depend on the context. The words may still be valid in some situations.
> 
> i agree on this one, it will become very difficult to have this strict check for the documents.

We always try to re-phrase such words in the documents, and we also spent a lot of time on the documentation to refactor them, I am not sure what hindering us from keeping an automatic check that we majorly comment on the PRs.

>  strict check for the documents.
So this is an optional check that can be ignored if some instances need it
thanks.
